client/blocks/README.md

```diff
@@ -0,0 +1,42 @@
+# Why we chose to use our pre-existing AJAX APIs instead of the WooCommerce Blocks APIs for Payment Request Buttons
+
+While we were in the process of migrating the Blocks integration from the WooCommerce Blocks plugin to Stripe we encountered several issues while adding support for Payment Requests with the Blocks API. The easiest way to fix these issues at the time was to use our pre-existing AJAX APIs and forego the Blocks API entirely.
+
+The WooCommerce Blocks API has since been significantly improved and we need to re-evaluate whether we can now use the Blocks API, see https://github.com/woocommerce/woocommerce-gateway-stripe/issues/1573.
+
+This file is meant to document why we chose to forego the Blocks API at the time, and what the downsides and benefits of that approach are.
+
+## What were the issues we encountered?
+
+-   3DS payments didn't work with saved cards and Payment Requests because some scripts weren't loaded in those contexts.
+-   The Payment Request Dialog [didn't close during 3DS validation](https://github.com/woocommerce/woocommerce-gateway-stripe/pull/1467#discussion_r623877272).
+-   State and zip code normalisation [didn't work](https://github.com/woocommerce/woocommerce-gateway-stripe/issues/1518) because we were missing some crucial normalisation functionality.
+
+Both of these have since been addressed in WooCommerce Blocks.
```

Comment:
The state and zip code normalization weren't addressed in WooCommerce Blocks, as far as I'm concerned. We can either remove this line or specify that only the 3DS issues were addressed.

Reply:
But we did address it. Or well, it was an issue when not using the AJAX APIs — issue in the sense that we'd have to duplicate the code. Just see https://github.com/woocommerce/woocommerce-gateway-stripe/issues/1518.

Or do you mean that because this is an issue Stripe has to solve we shouldn't include it here? I guess that's true, maybe including the fact that we don't have to duplicate this is all we should to say about this 🤔 

Please let me know what you think on that 🙏

Reply:
Ohhhh, sorry, I forgot to update the "Both of these have been addressed in WooCommerce Blocks" sentence 🤦‍♂️ 

You understood that correctly, I just forgot to update that 😬 

Hang on, I'll fix that ⌛

Reply:
I think part of the reason I interpreted it differently is because of the last part in the README, where it says the 3DS issues were fixed in the WooCommerce Blocks APIs.

> Another factor in this decision was that, during development, some parts of the WooCommerce Blocks API were unable to properly handle the required flow for the Payment Request Buttons, especially when it came to 3DS payments. This has since been fixed, and anyone else should consider using the WooCommerce Blocks APIs.

Reply:
Fixed in 707d35e399d3f492bb1e03044fecdabcabddb48f I think

Reply:
Maybe instead of "This has since been fixed, and anyone else should consider using the WooCommerce Blocks APIs." it would be better to say "The Blocks API has since been improved, and using the API should be the default choice for most projects"?

Reply:
Made ^ change in 325404fa09704c92813ca34bcfcd2bf25542a2c1

Reply:
> Made ^ change in 325404f

Looks good to me 👍